src/p2p/net_node.inl

```diff
@@ -847,6 +847,30 @@ namespace nodetool
   }
   //-----------------------------------------------------------------------------------
   template<class t_payload_net_handler>
+  bool node_server<t_payload_net_handler>::is_peer_used(const anchor_peerlist_entry& peer)
+  {
+    if(m_config.m_peer_id == peer.id) {
+        return true;//dont make connections to ourself
+    }
+
+    bool used = false;
+
+    m_net_server.get_config_object().foreach_connection([&](const p2p_connection_context& cntxt)
+    {
+      if(cntxt.peer_id == peer.id || (!cntxt.m_is_income && peer.adr.ip == cntxt.m_remote_ip && peer.adr.port == cntxt.m_remote_port))
```

Comment:
It should affect a peer from the anchor peer list the same way one from the white or gray. It can be removed to only match nodes by IP and port.

Am I missing something here?